pkg/uwb-dw1000/include/syscfg_uwb_dw1000.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_uwb-core
+ * @ingroup     uwb-core
+ * @{
+ *
+ * @file
+ * @brief       uwb-dw1000 module configurations
+ *              taken from decawave-uwb-dw1000/hw/drivers/uwb/uwb_dw1000
+ *
+ */
+
+#ifndef SYSCFG_UWB_DW1000_H
+#define SYSCFG_UWB_DW1000_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Used to check that the UWB_DEV has been setup correctly
+ *        at compile time.
+ * @note  (UWB_DEV_RXDIAG_MAXLEN > 19)
+ */
+#ifndef MYNEWT_VAL_UWB_DW1000_API_CHECKS
+#define MYNEWT_VAL_UWB_DW1000_API_CHECKS (1)
+#endif
+
+/**
+ * @brief Size of spi read/write buffer, sets the
+ *        maximum allowed nonblocking read operation
+ */
+#ifndef MYNEWT_VAL_DW1000_HAL_SPI_BUFFER_SIZE
+#define MYNEWT_VAL_DW1000_HAL_SPI_BUFFER_SIZE (256)
+#endif
+
+/**
+ * @brief The maximum number of bytes in a single transfer that the
+ *        SPI hardware supports. 255 is safe for nrf52.
```

Comment:
Uh if SPI peripherals have such limitations, we should amend the `periph/spi` API.
This can lead to some nasty surprises with SPI flash.

Reply:
This comes from upstream. might need to be adapted... but there are so many configurations that I started buy simply copying them as well as the defines.

Reply:
Fortunately the SPI driver already [takes care of it](https://github.com/RIOT-OS/RIOT/blob/master/cpu/nrf52/periph/spi.c#L174).

Reply:
Still needs to be defined for other platforms though, Should I adapt only the `255 is safe for nrf52` part?

Reply:
I mean it's not wrong that 255 is safe for nrf52 :wink: 
How is this define used?

I see the device can send up to 1kB frames - will the driver split up the transfer?

That should not be necessary if in RIOT the SPI API already takes care if this.

Reply:
The dw1000 hardware abstraction layers splits up the input data if larger than, similar to what is already done in the in RIOT spi API, but I'd rather not modify the pkg source code, even if that means some duplicate checks.

Reply:
We could use a higher value though and let the compiler throw it out.

Reply:
Hmmm for now it should not matter, it could eventually if we add an asynchronous spi. I'll increase the value and add a message on this

Reply:
addressed